service/controller/resource/azuremachinepoolconditions/conditionsubnetready.go

```diff
@@ -0,0 +1,111 @@
+package azuremachinepoolconditions
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-11-01/network"
+	azureconditions "github.com/giantswarm/apiextensions/v3/pkg/conditions/azure"
+	"github.com/giantswarm/microerror"
+	capzexp "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capiconditions "sigs.k8s.io/cluster-api/util/conditions"
+
+	"github.com/giantswarm/azure-operator/v5/pkg/helpers"
+)
+
+const (
+	subnetDeploymentPrefix     = "subnet"
+	provisioningStateSucceeded = "Succeeded"
+
+	SubnetNotFoundReason          = "SubnetNotFound"
+	SubnetProvisioningStatePrefix = "SubnetProvisioningState"
+)
+
+func (r *Resource) ensureSubnetReadyCondition(ctx context.Context, azureMachinePool *capzexp.AzureMachinePool) error {
+	r.logDebug(ctx, "ensuring condition %s", azureconditions.SubnetReadyCondition)
+
+	// Get Azure deployments client
+	deploymentsClient, err := r.azureClientsFactory.GetDeploymentsClient(ctx, azureMachinePool.ObjectMeta)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Now let's first check ARM deployment state
+	subnetDeploymentName := getSubnetDeploymentName(azureMachinePool.Name)
+	isSubnetDeploymentSuccessful, err := r.checkIfDeploymentIsSuccessful(ctx, deploymentsClient, azureMachinePool, subnetDeploymentName, azureconditions.SubnetReadyCondition)
+	if err != nil {
+		return microerror.Mask(err)
+	} else if !isSubnetDeploymentSuccessful {
+		// in the deployment is not yet successful, the check method will set
+		// appropriate condition value.
+		return nil
+	}
+
+	// Deployment is successful, now let's check the actual resource.
+	subnetsClient, err := r.azureClientsFactory.GetSubnetsClient(ctx, azureMachinePool.ObjectMeta)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	azureCluster, err := helpers.GetAzureClusterFromMetadata(ctx, r.ctrlClient, azureMachinePool.ObjectMeta)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	subnetName := azureMachinePool.Name
+	subnet, err := subnetsClient.Get(ctx, azureCluster.Name, azureCluster.Spec.NetworkSpec.Vnet.Name, subnetName, "")
+	if IsNotFound(err) {
+		r.setSubnetNotFound(ctx, azureMachinePool, subnetName, azureconditions.SubnetReadyCondition)
+		return nil
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Note: Here we check if the subnet exists and that its provisioning state
+	// is succeeded. It would be good to also check network security group,
+	// routing table and service endpoints.
+	if subnet.ProvisioningState == provisioningStateSucceeded {
```

Comment:
is there a constant in the SDK that we can use instead of our own `provisioningStateSucceeded`?

Reply:
There probably is, will look it up.